core/Widget/WidgetConfig.php

```diff
@@ -258,6 +259,10 @@ public function disable()
     public function checkIsEnabled()
     {
         if (!$this->isEnabled()) {
+            // Some widgets are disabled when the user is not superuser. If the user is not logged in, we should
+            // prompt them to do this first rather than showing them the "widget not enabled" error
+            Access::getInstance()->checkUserIsNotAnonymous();
```

Comment:
Wondering if that could make problems if a site is set as accessible for anonymous user, but e.g. the dashboard contains a widget that is disabled (for any kind of reason)

Reply:
Was thinking about that as well @sgiehl but it should be actually fine since it goes only in here when the widget is disabled. In this case the user would need to log in to see a more accurate message. We basically don't want to show that message then and also in general a user would actually not really get to see a disabled widget in the first place maybe (unless opened maybe directly not sure)